### DISCUSSION TITLE
Custom error handling with GraphQL

### DISCUSSION THREAD

**Discussion by lordvins226** [mswjs/msw#1331] [Q&A] [answered]
Is it possible to have errors in this format?

`"notFound": {
      "__typename": "NotFoundError",
      "message": "Entity 100 not found"
},`

**Comment by kettanaito** [comment, marked as answer]
Hi, @lordvins226. 

The code snippet you've posted doesn't look like a `GraphQLError` object. It looks like a regular field that represents an error. 

Yes, you can mock that kind of field with `ctx.data()`. 

```js
graphql.query('GetUser', (req, res, ctx) => {
  return res(
    ctx.data({
      notFound: {
        __typename: 'NotFoundError',
        message: 'Entity 100 not found',
      },
    }),
  )
})
```

Keep in mind what MSW describes the _server-side behavior_. Whichever payload you decide to return in the handler is what your client code will get. The actual error handling is always the client's responsibility.

**Reply by lordvins226** [reply]
Thanks @kettanaito
